### PR TITLE
feat: add setCollateral function

### PR DESCRIPTION
**Pull request by girazoki**
Co-authored-by: Mario Cao <mariocao@gmail.com>

### CODE REVIEW

**Comment by kelonye** [comment]
When merge? :)

**Comment by girazoki** [comment]
Once it is approved!

**Comment by kelonye** [comment]
I think the min collateral amount should be validated or is this something that could be changed in future?

Screenshot 2020-04-28 at 19 22 05

File: src/witnet/request.js [hunk 1]
```diff
@@ -15,6 +15,7 @@ class Request {
       commit_fee: 0,
       reveal_fee: 0,
       tally_fee: 0,
+      collateral: null,
```
Comment (aesedepece) [inline src/witnet/request.js]: As per WIP0002, this should default to the minimum collateral (`1_000_000_000`).

File: src/witnet/request.js [hunk 2]
```diff
@@ -48,6 +49,12 @@ class Request {
 
     return this
   }
+  setCollateral (collateral) {
+    if (collateral != 0){
```
Comment (aesedepece) [inline src/witnet/request.js]: Why guarding against `0`?

**Review by aesedepece — CHANGES_REQUESTED** [review]

File: src/witnet/request.js (hunk 2 above)
Comment (girazoki) [reply src/witnet/request.js]: Because the node was not able to serialize a 0. But I will guard against anything that is less than the minimum collateral

**Review by mariocao — APPROVED** [review]

File: src/witnet/request.js [hunk 3]
```diff
@@ -48,6 +49,12 @@ class Request {
 
     return this
   }
+  setCollateral (collateral) {
+    if (collateral < 1000000000){
+      this.data.collateral = collateral || this.data.collateral;
+    }
```
Comment (aesedepece) [inline src/witnet/request.js]: Shouldn't this be the other way?
```suggestion
    if (collateral > 1000000000){
      this.data.collateral = collateral;
    } else {
      this.data.collateral = 1000000000;
    }
```

**Review by aesedepece — CHANGES_REQUESTED** [review]

**Review by aesedepece — APPROVED** [review]

**Pull request merged**